File: P31/Clock/Clock.cs
using System.Security.Principal;
using CounterApp;
using SplashKitSDK;


namespace ClockApp
{
    public class Clock
    {
        private Counter _hour;
        private Counter _minute;
        private Counter _second;

        public Clock()
        {
            _hour = new Counter("Hour");
            _minute = new Counter("Minute");
            _second = new Counter("Second");
        }
        public void Tick()
        {
            IncrementSecond();
        }
        private void IncrementSecond()
        {
            _second.Increment();
            if (_second.Ticks == 60)
            {
                _second.Reset();
                IncrementMinute();
            }
        }
        private void IncrementMinute()
        {
            _minute.Increment();
            if (_minute.Ticks == 60)
            {
                _minute.Reset();
                IncrementHour();
            }
        }
        private void IncrementHour()
        {
            _hour.Increment();
            if (_hour.Ticks == 13)
            {
                _hour.Reset();
                _hour.Increment();
            }
        }

        public void Reset()
        {
            _hour.Reset();
            _minute.Reset();
            _second.Reset();
        }
        public string GetTime()
        {
            return $"{_hour.Ticks:D2}:{_minute.Ticks:D2}:{_second.Ticks:D2}";
        }
        //properties
        public string Hour
        {
         get
         {
                return _hour.Ticks.ToString("D2");
         }   
        }
        public string Minute
        {
            get
            {
                return _minute.Ticks.ToString("D2");
            }
        }
        public string Second
        {
            get
            {
                return _second.Ticks.ToString("D2");
            }
        }
    }
}

File: 11.1P/clock.py
from counter import Counter

class Clock:
    def __init__(self):
        self._hour = Counter("Hour")
        self._minute = Counter("Minute")
        self._second = Counter("Second")
    
    def tick(self):
        self._increment_second()
    
    def _increment_second(self):
        self._second.increment()
        if self._second.ticks == 60:
            self._second.reset()
            self._increment_minute()
    
    def _increment_minute(self):
        self._minute.increment()
        if self._minute.ticks == 60:
            self._minute.reset()
            self._increment_hour()
    
    def _increment_hour(self):
        self._hour.increment()
        if self._hour.ticks == 13:
            self._hour.reset()
            self._hour.increment()
    
    def reset(self):
        self._hour.reset()
        self._minute.reset()
        self._second.reset()
    
    def get_time(self):
        return f"{self._hour.ticks:02d}:{self._minute.ticks:02d}:{self._second.ticks:02d}"
    